cognition_objects/file_extraction.py

```diff
@@ -0,0 +1,113 @@
+from ..business_objects import general
+from ..session import session
+from ..models import FileExtraction
+from submodules.model import enums
+
+
+def get(
+    org_id: str,
+    file_reference_id: str,
+    extraction_key: str,
+    only_completed: bool = False,
+) -> FileExtraction:
+    return (
+        session.query(FileExtraction)
+        .filter(
+            FileExtraction.organization_id == org_id,
+            FileExtraction.file_reference_id == file_reference_id,
+            FileExtraction.extraction_key == extraction_key,
+            (
+                FileExtraction.state == enums.FileCachingState.COMPLETED.value
+                if only_completed
+                else True
+            ),
+        )
+        .first()
+    )
```

Comment:
usually we write this as a query extension since it's easier to read & easier to extend if needed

```suggestion

    query = session.query(FileExtraction)
        .filter(
            FileExtraction.organization_id == org_id,
            FileExtraction.file_reference_id == file_reference_id,
            FileExtraction.extraction_key == extraction_key,
        )
    
    if only_completed:
        query = query.filter( FileExtraction.state == enums.FileCachingState.COMPLETED.value)
    
    return query.first()
 
```